src/efftester.ml

```diff
@@ -290,275 +304,328 @@ exception No_solution
+      | Not_found -> VarSet.empty)
+      VarSet.empty
+  in
   VarSet.union concreteSet polySet
-
+;;
 
 (** Generators *)
 
 let a_code = int_of_char 'a'
 let z_code = int_of_char 'z'
 let alphaGen = Gen.map char_of_int (Gen.int_range a_code z_code)
 let varGen = Gen.map (String.make 1) alphaGen
-
 let stringGen = Gen.small_string ~gen:alphaGen
 let stringToString s = "\"" ^ s ^ "\""
-
 let sqrt i = int_of_float (sqrt (float_of_int i))
 
 let arb_int =
-  frequency
-    [(10,small_int);
-     (5,int);
-     (1, oneofl [min_int;-1;0;1;max_int])]
-  
+  frequency [ 10, small_int; 5, int; 1, oneofl [ min_int; -1; 0; 1; max_int ] ]
+;;
+
 let intGen = arb_int.gen (* Gen.small_int *)
 
 (* Type-directed literal generator *)
-let rec literalGen t eff size = match t with
-  | Unit   -> Gen.return LitUnit
-  | Int    -> Gen.map (fun i -> LitInt i) intGen
-  | Bool   -> Gen.map (fun b -> LitBool b) Gen.bool
+let rec literalGen t eff size =
+  match t with
+  | Unit -> Gen.return LitUnit
+  | Int -> Gen.map (fun i -> LitInt i) intGen
+  | Bool -> Gen.map (fun b -> LitBool b) Gen.bool
   | String -> Gen.map (fun s -> LitStr s) stringGen
   | List (Typevar _) -> Gen.return (LitList [])
   | List t ->
     if size = 0
     then Gen.return (LitList [])
-    else Gen.map (fun ls -> LitList ls)
-           (Gen.list_size (Gen.int_bound (sqrt size)) (literalGen t eff (sqrt size)))
-      (*     (Gen.list_size (Gen.int_bound (size/2)) (literalGen t eff (size/2))) *)
+    else
+      Gen.map
+        (fun ls -> LitList ls)
+        (Gen.list_size (Gen.int_bound (sqrt size)) (literalGen t eff (sqrt size)))
+  (*     (Gen.list_size (Gen.int_bound (size/2)) (literalGen t eff (size/2))) *)
   (* FIXME: - one element should/can have effect, if 'eff' allows *)
   (*        - list items should be able to contain arbitrary effectful exps *)
   | Typevar _ -> failwith "literalGen: typevar arg. should not happen"
-  | Fun _  -> failwith "literalGen: funtype arg. should not happen"
+  | Fun _ -> failwith "literalGen: funtype arg. should not happen"
+;;
 
-let effGen = Gen.oneofl [(false,false); (true,false)]
+let effGen = Gen.oneofl [ false, false; true, false ]
 
-let typeGen = (* Generates ground types (sans type variables) *)
-  Gen.fix
-    (fun recgen n ->
+let typeGen =
+  (* Generates ground types (sans type variables) *)
+  Gen.fix (fun recgen n ->
       if n = 0
-      then Gen.oneofl [Unit;Int;Bool;String]
-      else 
-	Gen.frequency [
-	(* Generate no alphas *)
-	  (4,Gen.oneofl [Unit;Int;Bool;String]);
-	  (1,Gen.map (fun t -> List t) (recgen (n/2)));
-	  (1,Gen.map3 (fun t e t' -> Fun (t,e,t')) (recgen (n/2)) effGen (recgen (n/2)));
-	])
-
-(* Sized generator of variables according to the LIT rule 
+      then Gen.oneofl [ Unit; Int; Bool; String ]
+      else
+        Gen.frequency
+          [ (* Generate no alphas *)
+            4, Gen.oneofl [ Unit; Int; Bool; String ]
+          ; 1, Gen.map (fun t -> List t) (recgen (n / 2))
+          ; ( 1
```

Comment:
I would prefer using `--break-separators=after`. My favorite list style is to have either the list on one line, or to have each bracket on its own line, and then one element per line, with a terminating `;` after each item, including the last. I believe this is the best style because it maximizes readability of diffs: adding or removing an item anywhere in the list includes just the new item in the diff. (With the present style, adding a new item at the front also adds a new `;` on the element that was previously in front, so the diff has two line; without the trailing separator, adding an item at the end has the same problem.)